lib/galaxy/model/migrations/README.md

```diff
@@ -1,115 +1,391 @@
-# Galaxy's Migrations System
+# Galaxy Database Schema Migrations
```

Comment:
The target audience is devs, maybe admins. I'll move this to a section for developers (or "advanced", or "internals", or something of that sort). It doesn't have to be pleasant reading - it must be helpful reading. It gives the minimum context that should help understand how it works, why there are 2 versions (one in the database, one in the codebase), and why they are expected to be the same. I've had this discussions on many occasions (explaining how it works, etc.), I it will help to have this as part of the docs (although moving it to a separate section seems to be the right way to do it)

Reply:
ok, this looks much better, can you please link it from the docs ?

Reply:
Will do!